### ISSUE TITLE
Format adds 14 seconds to output string

### ISSUE THREAD

**Issue by wubzz** [date-fns/date-fns#852]
I'm having difficulties understanding why 14 seconds is appended to the following output string when using `.format`. Unsure if this is a bug or not, perhaps I'm just missing something. So feel free to close if it's not.

```javsacript
dateFns.format('1899-12-30 00:00:00', 'YYYY-MM-DD HH:mm:ss');
//1899-12-30 00:00:14

dateFns.format(new Date('1899-12-30 00:00:00'), 'YYYY-MM-DD HH:mm:ss');
//1899-12-30 00:00:00
```

The former is wrong while the latter is correct. While I could use the latter, it doesn't work great in my case as I'm trying to allow any valid date-fns `input` including strings, so parsing internally rather than letting date-fns do it doesn't seem the right way to go.

**Comment by kossnocorp** [comment]
`1899-12-30T00:00:00` isn't a valid [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) string, so date-fns has a hard time parsing it. Starting with v2, date-fns will return you an invalid date for such string. You still can parse it using `new Date()` but I do not recommend it because it doesn't work consistently in different browsers. Consider using the ISO string or timestamps.

**Comment by kossnocorp** [comment]
It's unfortunate but it's in your interest. You don't want to one day figure out that the dates are all wrong in a specific browser that you never open.

You can use `parse` from v2 that allows parsing of any format: https://date-fns.org/v2.0.0-alpha.18/docs/parse

```js
parse('1899-12-30 00:00:00', 'yyyy-MM-dd hh:mmss')
```

**Comment by wubzz** [comment]
@kossnocorp Thanks but we have a policy not to include alpha/beta stage versions in our production application. I'll have to wait till it's stable. 

Also I tested the code in alpha18 and it results in `TypeError: 3 arguments required, but only 2 present`. No matter how I try to adjust the 3rd argument according to documentation I still get very weird results:

`0100-01-01 00:00:00` -> `0098-12-31T22:47:48.000Z`
`1899-12-30 00:00:00` -> `1899-01-29T22:59:46.000Z`

In my case it has nothing to do with browsers. They're dates stores in postgres updated and fetched via an API application in which we use formatters prior to sending the JSON to the client. The client may very well be a browser and may experience issues, of course.

I guess I just don't really understand the issue since `new Date('1899-12-30 00:00:00')` works perfectly fine. I was interested in how momentjs handles this since it works there, so I compared date-fns to momentjs _(even if it's apples and oranges)_ and moment seems to accept this as an ISO string. (They check ISO first, RFC 2822 second, new Date(input) final)
